capstone/src/main/webapp/js/autocomplete.js

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable no-unused-vars */
+
+/*
+ * Largely inspired by JS code from W3Schools: https://www.w3schools.com/howto/howto_js_autocomplete.asp
+ */
+const SEARCH_INPUT = document.getElementById('search');
+const ITEM_CLASS = 'autocomplete-items';
+const ACTIVE_CLASS = 'autocomplete-active';
+const SUGGESTION_LIST_ID = 'autocomplete-list';
+const UP_KEY = 'ArrowUp';
+const DOWN_KEY = 'ArrowDown';
+const ENTER_KEY = 'Enter';
+let selectedSuggestionPosition = -1;
+
+/**
+ * Adds the 'active' tag to an autocomplete elmt
+ * @param {HTMLDivElement} autocompleteListElmt
+ */
+function addActiveTag(autocompleteListElmt) {
+  if (!autocompleteListElmt) {
+    return;
+  }
+  removeActiveTag(autocompleteListElmt);
+  if (selectedSuggestionPosition >= autocompleteListElmt.length) {
+    selectedSuggestionPosition = 0;
+  } else if (selectedSuggestionPosition < 0) {
+    selectedSuggestionPosition = (autocompleteListElmt.length - 1);
+  }
+  autocompleteListElmt[selectedSuggestionPosition].classList.add(ACTIVE_CLASS);
+}
+
+/**
+ * Removes the 'active' tag from an autocomplete elmt
+ * @param {HTMLDivElement} autocompleteListElmt
+ */
+function removeActiveTag(autocompleteListElmt) {
+  for (elmt of autocompleteListElmt) {
+    elmt.classList.remove(ACTIVE_CLASS);
+  }
+}
+
+/**
+ * Closes dropdown autocomplete list
+ * Does not close the element provided
+ * @param {HTMLDivElement} elmnt
+ */
+function closeAllElmntExcept(elmnt) {
+  const autoItems = document.getElementsByClassName(ITEM_CLASS);
+  for (const item of autoItems) {
+    if (item != elmnt) {
+      item.parentNode.removeChild(item);
+    }
+  }
+}
+
+/**
+ * Adds valid suggestions to list of stored suggestions
+ * @param {string} validSuggestion
+ */
+function keepTrackOfSuggestions(validSuggestion) {
+  const parent = document.getElementById('suggestions');
+  const listElt = document.createElement('li');
+  listElt.innerHTML = validSuggestion;
+  parent.appendChild(listElt);
+}
+
+/**
+ * Current placeholder until servlet is created
+ * @return {Array<string>} arr
+ */
+function getListOfSuggestions() {
+  const arr = ['Hallo', 'Hello', 'Hi', 'Hiya', 'Howdy', 'Wassup'];
+  return arr;
+}
+
+/**
+ * Identifies and returns relevant suggestions in the arr
+ * @param {string[]} arr
```

Comment:
This should be `Array<string>` instead of string[]
Same comment below

Reply:
thanks! done!